### PR TITLE
refactor(Tree): Make skip marks objects

### PR DESCRIPTION
**Pull request by yann-achard-MS**
Refactors the marks that represent skipped nodes so that they are more similar to the other types of marks.
This makes the format more homogenous, which reduces code clutter and avoids type kludges (see the changes to `isExistingCellMark`).

### CODE REVIEW

File: experimental/dds/tree2/src/feature-libraries/sequence-field/format.ts [hunk 1]
```diff
@@ -12,7 +12,24 @@ import { ChangesetLocalId, NodeChangeset } from "../modular-schema";
...
 
 export type NodeCount = number;
-export type Skip = number;
+
+/**
+ * Left undefined for terseness.
+ */
+export const SkipType = undefined;
+
+export interface Skip extends CellTargetingMark {
```
Comment (alex-pardes) [inline experimental/dds/tree2/src/feature-libraries/sequence-field/format.ts:21]: While it's fine for there to be a skip mark targeting a detached cell, it is unnecessary for us to ever create them.

Comment (alex-pardes) [inline experimental/dds/tree2/src/feature-libraries/sequence-field/format.ts:21]: What do you think about renaming `Skip` to something like `UnchangedMark` or `PlaceholderMark`?

File: experimental/dds/tree2/src/feature-libraries/sequence-field/sequenceFieldToDelta.ts [hunk 2]
```diff
@@ -29,9 +23,9 @@ export function sequenceFieldToDelta<TNodeChange>(
 			out.push(deltaFromNodeChange(getNodeChange(mark), getMarkLength(mark), deltaFromChild));
 		} else if (areInputCellsEmpty(mark) && areOutputCellsEmpty(mark)) {
 		} else {
-			assert(isObjMark(mark), "Cell changing mark must be an ObjMark");
 			// Inline into `switch(mark.type)` once we upgrade to TS 4.7
 			const type = mark.type;
+			assert(type !== SkipType, "Cell changing mark must be an ObjMark");
```
Comment (alex-pardes) [inline experimental/dds/tree2/src/feature-libraries/sequence-field/sequenceFieldToDelta.ts:28]:
```suggestion
			assert(type !== SkipType, "Cell changing mark must not be a skip");
```

**Review by alex-pardes — APPROVED** [review]

File: experimental/dds/tree2/src/feature-libraries/sequence-field/format.ts (hunk 1 above)
Comment (yann-achard-MS) [reply experimental/dds/tree2/src/feature-libraries/sequence-field/format.ts:21]: I considered add these fields to capture that in the type system
```
	detachEvent?: undefined;
	lineage?: undefined;
```
but when we refactor to orthogonalize the where/anchor from the what/effect we'll likely tolerate such a thing anyway.

Comment (yann-achard-MS) [reply experimental/dds/tree2/src/feature-libraries/sequence-field/format.ts:21]: How about `NoopMark`?

Comment (alex-pardes) [reply experimental/dds/tree2/src/feature-libraries/sequence-field/format.ts:21]: Sure

**Pull request merged**